File: SDK Project/app/src/main/java/com/cardgame/sdk/data/PlayerData.java
package com.cardgame.sdk.data;

import java.util.ArrayList;
import java.util.Collections;

public class PlayerData {
    public ArrayList<CardData> Deck;
    public HealthData HealthData;

    public PlayerData(CardData[] deck, int maxHealth) {
        this.Deck = new ArrayList<>();
        Collections.addAll(this.Deck, deck);

        this.HealthData = new HealthData(maxHealth);
    }
}

File: Server Project/SharedLibrary/PlayerData.cs
using System.Collections.Generic;
using System.Linq;

namespace CardGame.Data
{
    [System.Serializable]
    public class PlayerData
    {
        public List<CardData> Deck;
        public HealthData HealthData;

        public PlayerData(CardData[] deck, int maxHealth)
        {
            Deck = deck.ToList();
            HealthData = new HealthData(maxHealth);
        }

        public bool HasCards() => Deck?.Count > 1;
    }
}
